From f4d98ff79d5a79102b777ea7e23002d9f7326489 Mon Sep 17 00:00:00 2001
From: Paul Floyd <pjfloyd@wanadoo.fr>
Date: Tue, 8 Dec 2020 15:07:29 +0100
Subject: [PATCH] Bug 408663 - Patch: Suppression file for musl libc

Origin: Upstream
Upstream-Status: Backport
[ismael@iodev.co.uk: For some reason the file is missing in the tarball]
---
 musl.supp    | 46 ++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 46 insertions(+)
 create mode 100644 musl.supp

diff --git a/musl.supp b/musl.supp
new file mode 100644
index 000000000..864172a24
--- /dev/null
+++ b/musl.supp
@@ -0,0 +1,46 @@
+# Suppressions for musl libc
+# See: https://www.openwall.com/lists/musl/2017/06/15/4
+
+{
+   musl-dynlink-false-positive1
+   Memcheck:Leak
+   fun:calloc
+   fun:load_direct_deps
+   fun:load_deps
+   fun:load_deps
+   fun:__dls3
+   fun:__dls2b
+   fun:__dls2
+}
+
+{
+   musl-dynlink-false-positive2
+   Memcheck:Leak
+   fun:calloc
+   fun:load_direct_deps
+   fun:load_deps
+   fun:load_deps
+   fun:__dls3
+   fun:__dls2
+}
+
+{
+   musl-dynlink-false-positive3
+   Memcheck:Leak
+   fun:calloc
+   fun:load_library
+   fun:load_preload
+   fun:__dls3
+   fun:__dls2b
+   fun:__dls2
+}
+
+{
+   musl-dynlink-false-positive4
+   Memcheck:Leak
+   fun:calloc
+   fun:load_library
+   fun:load_preload
+   fun:__dls3
+   fun:__dls2
+}
